src/GraphQLParser/Visitors/SDLPrinter.cs

```diff
@@ -1097,13 +1097,15 @@ public SDLPrinter(SDLPrinterOptions options)
     }
 
     /// <inheritdoc cref="SDLPrinter{TContext}"/>
-    public virtual ValueTask PrintAsync(ASTNode node, TextWriter writer, CancellationToken cancellationToken = default)
+    public virtual async ValueTask PrintAsync(ASTNode node, TextWriter writer, CancellationToken cancellationToken = default)
     {
         var context = new DefaultPrintContext(writer)
         {
             CancellationToken = cancellationToken,
         };
-        return VisitAsync(node, context);
+        await VisitAsync(node, context).ConfigureAwait(false);
+        if (!context.NewLinePrinted && node is GraphQLDocument)
+            await writer.WriteLineAsync().ConfigureAwait(false);
```

Comment:
Changes were made only within `PrintAsync` and only for printed documents, not for individual nodes being printed, or where `VisitAsync` was called directly and the context could be examined to determine whether or not a newline had been printed.